src/main/java/org/apache/commons/text/CaseUtils.java

```diff
@@ -0,0 +1,140 @@
+ *
+ * @since 1.0
+ */
+public class CaseUtils {
+
+    /**
+     * <p><code>CaseUtils</code> instances should NOT be constructed in
+     * standard programming. Instead, the class should be used as
+     * <code>CaseUtils.toCamelCase("foo bar", true, new char[]{'-'});</code>.</p>
+     * <p>
+     * <p>This constructor is public to permit tools that require a JavaBean
+     * instance to operate.</p>
+     */
+    public CaseUtils() {
+        super();
+    }
+
+    // Camel Case
+    //-----------------------------------------------------------------------
+    /**
+     * <p>Converts all the delimiter separated words in a String into camelCase,
+     * that is each word is made up of a titlecase character and then a series of
+     * lowercase characters. The  </p>
+     *
+     * <p>The delimiters represent a set of characters understood to separate words.
+     * The first non-delimiter character after a delimiter will be capitalized. The first String
+     * character may or may not be capitalized and it's determined by the user input for capitalizeFirstLetter
+     * variable.</p>
+     *
+     * <p>A <code>null</code> input String returns <code>null</code>.
+     * Capitalization uses the Unicode title case, normally equivalent to
+     * upper case.</p>
+     *
+     * <pre>
+     * CaseUtils.toCamelCase(null, false)                                 = null
+     * CaseUtils.toCamelCase("", false, *)                                = ""
+     * CaseUtils.toCamelCase(*, false, null)                              = *
+     * CaseUtils.toCamelCase(*, true, new char[0])                        = *
+     * CaseUtils.toCamelCase("To.Camel.Case", false, new char[]{'.'})     = "toCamelCase"
+     * CaseUtils.toCamelCase(" to @ Camel case", true, new char[]{'@'})   = "toCamelCase"
+     * CaseUtils.toCamelCase(" @to @ Camel case", false, new char[]{'@'}) = "toCamelCase"
+     * </pre>
+     *
+     * @param str  the String to be converted to camelCase, may be null
+     * @param capitalizeFirstLetter boolean that determines if the first character of first word should be title case.
+     * @param delimiters  set of characters to determine capitalization, null and/or empty array means whitespace
+     * @return camelCase of String, <code>null</code> if null String input
+     */
+    public static String toCamelCase(String str, boolean capitalizeFirstLetter, final char... delimiters) {
+        if (StringUtils.isEmpty(str)) {
+            return str;
+        }
+        str = str.toLowerCase();
+        int strLen = str.length();
+        int [] newCodePoints = new int[strLen];
+        int outOffset = 0;
+        Set<Integer> delimiterSet = generateDelimiterSet(delimiters);
+        boolean capitalizeNext = false;
+        if (capitalizeFirstLetter) {
+            capitalizeNext = true;
+        }
+        for (int index = 0; index < strLen;) {
+            final int codePoint = str.codePointAt(index);
```

Comment:
If we decide to go with `CharSequence` here, this becomes a tougher move. We have to begin worrying about UTF-16 and "surrogate pairs' of characters. See [docs here](https://docs.oracle.com/javase/8/docs/api/java/lang/Character.html#unicode)